homeassistant/components/apple_tv/__init__.py

```diff
@@ -229,7 +225,12 @@ async def _connect_loop(self):
                 if conf:
                     await self._connect(conf)
             except exceptions.AuthenticationError:
-                self._auth_problem()
+                self.config_entry.async_start_reauth(self.hass)
+                asyncio.create_task(self.disconnect())
```

Comment:
Should we make sure that we are disconnected before we start the re-auth flow to avoid multiple connections at the same time? Does it matter?

Reply:
Doesn't matter, connection will closed long before re-auth starts and having multiple connections open is not a problem either.